Source/FakeItEasy.Specs/DisposableSpecs.cs

```diff
@@ -0,0 +1,55 @@
+﻿namespace FakeItEasy.Specs
+{
+    using System;
+    using FluentAssertions;
+    using Machine.Specifications;
+
+    public class when_faking_a_disposable_class
+        : EventRaisingSpecs
+    {
+        static IDisposable fake;
+        private static Exception exception;
+
+        Establish context = () =>
+        {
+            AppDomain.CurrentDomain.UnhandledException += UnhandledExceptionHandler;
+
+            fake = A.Fake<SomeDisposable>();
+        };
+
+        Because of = () =>
+        {
+            fake = null;
+            GC.Collect();
+            GC.WaitForPendingFinalizers();
+        };

```

Comment:
Does this fail without the fix? I guess it probably does, but the reason I put in the extra collects in https://github.com/FakeItEasy/FakeItEasy/issues/440#issuecomment-71342374 was to ensure that the deallocation of everything else happens *before* the finalization of the fake. Of course, `GC.Collect()` is, contrary to popular belief, only a *hint* to the GC that it can perform collection so I guess this doesn't make a lot of difference anyway.